activerecord/lib/active_record/table_metadata.rb

```diff
@@ -12,13 +12,13 @@ def initialize(klass, arel_table, association = nil)
     def resolve_column_aliases(hash)
       # This method is a hot spot, so for now, use Hash[] to dup the hash.
       #   https://bugs.ruby-lang.org/issues/7166
-      hash = Hash[hash]
-      hash.keys.grep(Symbol) do |key|
-        if klass.attribute_alias? key
-          hash[klass.attribute_alias(key)] = hash.delete key
+      new_hash = Hash[hash]
```

Comment:
Since we are looping through all the hash, does it make difference if we build a new hash and copy the values to it?

```ruby
new_hash = {}
hash.each do |key, value|
  if (key.is_a?(Symbol)) && klass.attribute_alias?(key)
    new_hash[klass.attribute_alias(key)] = new_hash.delete(key)
  else
    new_hash[key] = value
  end
end
```